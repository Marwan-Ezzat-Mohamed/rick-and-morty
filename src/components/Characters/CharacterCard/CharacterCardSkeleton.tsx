import Card from '@mui/material/Card';
import CardContent from '@mui/material/CardContent';
import Skeleton from '@mui/material/Skeleton';
import Typography from '@mui/material/Typography';
import Box from '@mui/material/Box';
import { useTheme } from '@mui/material/styles';
import { tokens } from './../../../theme';
import styles from './styles';
import { memo } from 'react';

const CharacterCardSkeleton = () => {
  const theme = useTheme();
  const colors = tokens(theme.palette.mode);

  const Information = () => {
    return (
      <Box>
        <Box
          sx={{
            display: 'flex',
            color: '#8f949f',
            alignItems: 'center',
          }}
        >
          <Typography variant="h5" width="40%">
            <Skeleton />
          </Typography>
        </Box>
        <Typography variant="h5">
          <Skeleton />
        </Typography>
      </Box>
    );
  };

  return (
    <Card
      sx={{
        ...styles.card,
        backgroundColor: colors.primary[400],
        '&:hover': {
          backgroundColor: colors.primary[500],
          transform: 'scale(1.05)',
        },
      }}
    >
      <Skeleton variant="rectangular" sx={styles.media} />
      <CardContent sx={styles.content}>
        <Typography variant="h3" fontWeight="bold">
          <Skeleton />
        </Typography>

        {Information()}
        {Information()}
        {Information()}
      </CardContent>
    </Card>
  );
};

export default memo(CharacterCardSkeleton);
