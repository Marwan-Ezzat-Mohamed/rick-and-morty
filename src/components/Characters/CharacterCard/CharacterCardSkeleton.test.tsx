import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import CharacterCardSkeleton from './CharacterCardSkeleton';

describe('CharacterCardSkeleton', () => {
  it('matches snapshot', () => {
    const rendered = render(<CharacterCardSkeleton />);
    expect(rendered).toMatchSnapshot();
  });
});
