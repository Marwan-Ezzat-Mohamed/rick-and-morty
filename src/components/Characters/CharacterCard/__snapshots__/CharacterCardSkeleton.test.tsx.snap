// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterCardSkeleton matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-dwp3qi-MuiPaper-root-MuiCard-root"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-12ect3d-MuiSkeleton-root"
        />
        <div
          class="MuiCardContent-root css-1wkgrvo-MuiCardContent-root"
        >
          <h3
            class="MuiTypography-root MuiTypography-h3 css-1ey9xtw-MuiTypography-root"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
            />
          </h3>
          <div
            class="MuiBox-root css-0"
          >
            <div
              class="MuiBox-root css-11rx8v9"
            >
              <h5
                class="MuiTypography-root MuiTypography-h5 css-fzjimn-MuiTypography-root"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
                />
              </h5>
            </div>
            <h5
              class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
              />
            </h5>
          </div>
          <div
            class="MuiBox-root css-0"
          >
            <div
              class="MuiBox-root css-11rx8v9"
            >
              <h5
                class="MuiTypography-root MuiTypography-h5 css-fzjimn-MuiTypography-root"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
                />
              </h5>
            </div>
            <h5
              class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
              />
            </h5>
          </div>
          <div
            class="MuiBox-root css-0"
          >
            <div
              class="MuiBox-root css-11rx8v9"
            >
              <h5
                class="MuiTypography-root MuiTypography-h5 css-fzjimn-MuiTypography-root"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
                />
              </h5>
            </div>
            <h5
              class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
              />
            </h5>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-dwp3qi-MuiPaper-root-MuiCard-root"
    >
      <span
        class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-12ect3d-MuiSkeleton-root"
      />
      <div
        class="MuiCardContent-root css-1wkgrvo-MuiCardContent-root"
      >
        <h3
          class="MuiTypography-root MuiTypography-h3 css-1ey9xtw-MuiTypography-root"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
          />
        </h3>
        <div
          class="MuiBox-root css-0"
        >
          <div
            class="MuiBox-root css-11rx8v9"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5 css-fzjimn-MuiTypography-root"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
              />
            </h5>
          </div>
          <h5
            class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
            />
          </h5>
        </div>
        <div
          class="MuiBox-root css-0"
        >
          <div
            class="MuiBox-root css-11rx8v9"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5 css-fzjimn-MuiTypography-root"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
              />
            </h5>
          </div>
          <h5
            class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
            />
          </h5>
        </div>
        <div
          class="MuiBox-root css-0"
        >
          <div
            class="MuiBox-root css-11rx8v9"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5 css-fzjimn-MuiTypography-root"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
              />
            </h5>
          </div>
          <h5
            class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
            />
          </h5>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
